dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AbstractGistReadOnlyController.java

```diff
@@ -199,6 +208,17 @@ private static GistQuery createGistQuery( HttpServletRequest request,
             new JsonBuilder( jsonMapper ).skipNullMembers().toObject( gistService.describe( query ) ) );
     }
 
+    private void gistToCsvResponse( HttpServletResponse response, GistQuery query )
+        throws IOException
+    {
+        query = gistService.plan( query ).toBuilder().references( false ).build();
+        response.addHeader( "Content-Type", "text/csv" );
```

Comment:
Could use `org.springframework.http.HttpHeaders` `CONTENT_TYPE` here for "type safety".

Reply:
done